src/formatDate.js

```diff
@@ -51,6 +51,53 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const result = [];
+
+  const fromSeparator = fromFormat.pop();
+  const inputDateArray = date.split(fromSeparator);
+  const newSeparator = toFormat.pop();
```

Comment:
Don't mutate incoming arrays.
```suggestion
  const fromSeparator = [...fromFormat].pop();
  const inputDateArray = date.split(fromSeparator);
  const newSeparator = [...toFormat].pop();
```